### PR TITLE
SCP-2166: Disabling god mode and determining roles from wallet assets

### PR DESCRIPTION
**Pull request by merivale**
This feels a bit ugly, but it works. To make it less ugly we probably need to rethink some of the types (notably `MarloweParams`) and add some lenses for the `Marlowe.PAB` things.

### CODE REVIEW

File: marlowe-dashboard-client/src/Contract/View.purs [hunk 1]
```diff
@@ -684,3 +702,23 @@ getParty (IDeposit _ p _ _) = Just p
 getParty (IChoice (ChoiceId _ p) _) = Just p
 
 getParty _ = Nothing
+
+isParty :: WalletDetails -> MarloweParams -> Party -> Boolean
+isParty walletDetails marloweParams party = Set.member party $ getParties walletDetails marloweParams
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Contract/View.purs:707]: isParty seems to be called a lot and I'm not sure if it changes (or if it changes that much), we might want to have `getParties` cached in the contract state, right?

**Review by hrajchert — APPROVED** [review]
LGTM, just a minor concern about performance in comment

File: marlowe-dashboard-client/src/Contract/View.purs (hunk 1 above)
Comment (merivale) [reply marlowe-dashboard-client/src/Contract/View.purs:707]: Good point. In the real thing, we might let people give their roles to others. But for now at least, it's never going to change.

**Comment by merivale** [comment]
I'm leaving god mode enabled for now - until we get the PAB stuff working properly.

**Pull request merged**